### PR TITLE
[TTS] Add VietnameseCharsTokenizer

### PR DESCRIPTION
**Pull request by huutuongtu**
Signed-off-by: Tu [huutu12312vn@gmail.com](mailto:huutu12312vn@gmail.com)

# What does this PR do ?

Add a Vietnamese language tokenizer for TTS training

**Collection**: [TTS]

# Changelog 
- Add VietnameseCharsTokenizer
- Add unit tests for Vietnamese

# Usage
```python
from nemo.collections.common.tokenizers.text_to_speech.tts_tokenizers import VietnameseCharsTokenizer

text = "Xin chào các bạn."

tokenizer = VietnameseCharsTokenizer(
    pad_with_space=True,
)

tokens = tokenizer(text)
graphemes = tokenizer.decode(tokens)
graphemes = graphemes.replace('|', '')

print(tokens)
#  xin chào các bạn.
print(graphemes)
```

# Before your PR is "Ready for review"
**Pre checks**:
- [x] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [x] Did you write any new necessary tests?
- [x] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [x] New Feature
- [ ] Bugfix
- [ ] Documentation

### CODE REVIEW

File: nemo/collections/common/tokenizers/text_to_speech/tts_tokenizers.py [hunk 1]
```diff
@@ -203,6 +204,45 @@ def __init__(
         )
 
 
+class VietnameseCharsTokenizer(BaseCharsTokenizer):
+
+    _punctuation = '-();:,.!?¡¿—…"«»“”%'
+    _letters = 'abcdefghijklmnopqrstuvwxyzđáàãảạăắằẵẳặâấầẫẩậóòõỏọôốồỗổộơớờỡởợéèẽẻẹêếềễểệúùũủụưứừữửựíìĩỉịýỳỹỷỵ'
```
Comment (XuesongYang) [inline nemo/collections/common/tokenizers/text_to_speech/tts_tokenizers.py:210]: could you please move `_punctuation` and `_letters` to `nemo/collections/common/tokenizers/text_to_speech/ipa_lexicon.py` accordingly, just following the same convention as `GermanCharsTokenizer`?

**Review by XuesongYang — CHANGES_REQUESTED** [review]
please refactor the code accordingly.

File: nemo/collections/common/tokenizers/text_to_speech/ipa_lexicon.py [hunk 2]
```diff
@@ -184,6 +208,32 @@ def get_ipa_punctuation_list(locale):
                 '—',  # em dash, U+2014, decimal 8212
             ]
         )
+    if locale == "vi-VN":
+        punct_set.update(
```
Comment (XuesongYang) [inline nemo/collections/common/tokenizers/text_to_speech/ipa_lexicon.py:212]: could you pls add the source of punctuations of Vietnamese?

**Review by XuesongYang — DISMISSED** [review]
thanks. LGTM. pls add the source of Vietnamese punctuations if any.

File: nemo/collections/common/tokenizers/text_to_speech/ipa_lexicon.py (hunk 2 above)
Comment (huutuongtu) [reply nemo/collections/common/tokenizers/text_to_speech/ipa_lexicon.py:212]: Hmm, it seems that there isn't any 'official' source that talks about Vietnamese punctuation marks. I can find some information about punctuation marks here: https://languagedrops.com/word/en/english/vietnamese/topics/punctuation/. 
Maybe we just need to use DEFAULT_PUNCTUATION.

**Review by XuesongYang — DISMISSED** [review]

**Review by XuesongYang — APPROVED** [review]

**Pull request merged**